### DISCUSSION TITLE
Ordering by foreign key count

### DISCUSSION THREAD

**Discussion by brycedorn** [supabase/supabase#7875] [Questions] [answered]
Hi, is it possible to do something like this:

```js
await supabase.from("groups").select('*, items(id)').order('items.count');
```

Basically, sort resulting groups by the number of items tied to that group (`group_id === id`). Ideally I'd like to avoid keeping a column to track `items.count` and instead let `order` handle it.

**Comment by steve-chavez** [comment, marked as answer]
Not possible directly for now but you can do so with a computed column.

```sql
create function items_count(groups) returns bigint as $$
  select count(*) from items where group_id = $1.id;
$$ stable language sql;
```

```js
await supabase.from("groups").select('*, items(id)').order('items_count');
```

**Reply by brycedorn** [reply]
Thank you @steve-chavez! I wasn't sure how to structure/call the function, this works perfectly.

**Reply by devalensio** [reply]
@steve-chavez where do we write the items_count (computed column) in our supabase dashboard ?

**Reply by steve-chavez** [reply]
@devalensio Since it's just a SQL function, you can use the UI SQL editor.

I think the UI Function Editor will not work now because it doesn't detect table type arguments(you could request this is as a feature).

**Reply by devalensio** [reply]
@steve-chavez thanks 🙏🏼

**Reply by hasanneke** [reply]
@steve-chavez Amazing, thanks!
